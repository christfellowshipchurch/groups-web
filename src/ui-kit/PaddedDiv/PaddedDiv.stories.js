import React from 'react';
import { storiesOf } from '@storybook/react';

import PaddedDiv from '.';

storiesOf('ui-kit/PaddedDiv', module)
  .addDecorator((story) => (
    <div style={{ width: '50%', background: 'salmon' }}>{story()}</div>
  ))
  .add('default', () => <PaddedDiv>This text is in a PaddedDiv</PaddedDiv>)
  .add('horizontal (false)', () => (
    <PaddedDiv horizontal={false}>
      This text is in a PaddedDiv without horizontal padding
    </PaddedDiv>
  ))
  .add('vertical (false)', () => (
    <PaddedDiv vertical={false}>
      This text is in a PaddedDiv without vertical padding
    </PaddedDiv>
  ));
