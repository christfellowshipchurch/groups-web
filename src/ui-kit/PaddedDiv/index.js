import styled from 'styled-components';
import PropTypes from 'prop-types';

import { withOverrides } from '../theme';

const PaddedDiv = styled.div.attrs(withOverrides('ui-kit-PaddedDiv'))`
  padding: ${({ horizontal, theme, vertical }) => {
    const verticalValue = vertical ? theme.sizing.baseUnit(1) : 0;
    const horizontalValue = horizontal ? theme.sizing.baseUnit(1) : 0;

    return `${verticalValue} ${horizontalValue}`;
  }};
`;

PaddedDiv.propTypes = {
  horizontal: PropTypes.bool,
  vertical: PropTypes.bool,
};

PaddedDiv.defaultProps = {
  horizontal: true,
  vertical: true,
};

PaddedDiv.displayName = 'ui-kit.PaddedDiv';

export default PaddedDiv;
