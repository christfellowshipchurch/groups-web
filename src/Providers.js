import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
