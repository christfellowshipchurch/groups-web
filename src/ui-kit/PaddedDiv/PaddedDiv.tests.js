import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../Providers';

import PaddedDiv from '.';

describe('the PaddedDiv component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <PaddedDiv />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render without horizontal padding', () => {
    const tree = renderer.create(
      <Providers>
        <PaddedDiv horizontal={false} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render without vertical padding', () => {
    const tree = renderer.create(
      <Providers>
        <PaddedDiv vertical={false} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
