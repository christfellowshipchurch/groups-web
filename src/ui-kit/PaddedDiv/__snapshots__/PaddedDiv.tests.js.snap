// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PaddedDiv component should render 1`] = `
.c0 {
  padding: 1rem 1rem;
}

<div
  className="c0 ui-kit-PaddedDiv"
/>
`;

exports[`the PaddedDiv component should render without horizontal padding 1`] = `
.c0 {
  padding: 1rem 0;
}

<div
  className="c0 ui-kit-PaddedDiv"
/>
`;

exports[`the PaddedDiv component should render without vertical padding 1`] = `
.c0 {
  padding: 0 1rem;
}

<div
  className="c0 ui-kit-PaddedDiv"
/>
`;
